source/common/stats/stat_merger.cc

```diff
@@ -78,32 +68,23 @@ void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges)
     parent_value_ref = new_parent_value;
 
     auto& gauge_ref = target_store_.gauge(gauge.first);
-    Gauge::CombineLogic combine_logic = StatMerger::getCombineLogic(gauge_ref, gauge.first);
-    if (combine_logic == Gauge::CombineLogic::NoImport) {
+    if (!StatMerger::shouldImport(gauge_ref, gauge.first)) {
       continue;
     }
-    // If undefined, take parent's value unless explicitly told NoImport.
-    if (!gauge_ref.used()) {
-      gauge_ref.set(new_parent_value);
-      continue;
-    }
-    switch (combine_logic) {
-    case Gauge::CombineLogic::OnlyImportWhenUnusedInChild:
-      // Already set above; nothing left to do.
-      break;
-    case Gauge::CombineLogic::NoImport:
-      break;
-    case Gauge::CombineLogic::Accumulate:
-      if (new_parent_value > old_parent_value) {
-        gauge_ref.add(new_parent_value - old_parent_value);
-      } else {
-        gauge_ref.sub(old_parent_value - new_parent_value);
-      }
-      break;
+    if (new_parent_value > old_parent_value) {
+      gauge_ref.add(new_parent_value - old_parent_value);
+    } else {
+      gauge_ref.sub(old_parent_value - new_parent_value);
     }
   }
 }
 
+// TODO(fredlas) the current implementation can "leak" obsolete parent stats into the child.
```

Comment:
Do you plan on working on this as an immediate follow up? If not should we open a tracking issue right now? I would like to see this done sooner rather than later?

Reply:
I took a quick look through the relevant code, and yes, this can actually be a quite quick and easy followup. I'll do it very soon.

Reply:
Oh wait.... I thought you were referring to the pagination here. This one I think is also pretty straightforward, though, if you think the flag bit approach makes sense.

Reply:
I responded above, let's figure this out in a follow up. IMO it's much more important to fix the leak vs. the pagination if we are talking about priority. I don't think the pagination is really going to be a huge deal now that you are only sending counters that have data in them.